e2e/test/iothub/ConnectionStatusChangeHandlerTests.cs

```diff
@@ -14,32 +14,32 @@ namespace Microsoft.Azure.Devices.E2ETests
     [TestCategory("IoTHub")]
     public class ConnectionStatusChangeHandlerTests : E2EMsTestBase
     {
-        private readonly string DevicePrefix = $"{nameof(ConnectionStatusChangeHandlerTests)}_Device";
+        private readonly string DevicePrefix = $"{nameof(ConnectionStatusChangeHandlerTests)}Disconnected";
         private readonly string ModulePrefix = $"{nameof(ConnectionStatusChangeHandlerTests)}";
 
         [LoggedTestMethod, Timeout(ConnectionStateChangeTestTimeoutMilliseconds)]
         [TestCategory("LongRunning")]
-        public async Task IotHubDeviceClient_DeviceDeleted_Gives_ConnectionStatus_DeviceDisabled_AmqpTcp()
+        public async Task IotHubDeviceClient_DeviceDeleted_Gives_ConnectionStatus_Disconnected_ChangeReason_DeviceDisabled_AmqpTcp()
         {
-            await IotHubDeviceClient_Gives_ConnectionStatus_DeviceDisabled_Base(
+            await IotHubDeviceClient_Gives_ConnectionStatus_Disconnected_ChangeReason_DeviceDisabled_Base(
                 new IotHubClientAmqpSettings(IotHubClientTransportProtocol.Tcp),
                 async (r, d) => await r.Devices.DeleteAsync(d).ConfigureAwait(false)).ConfigureAwait(false);
         }
 
         [TestCategory("LongRunning")]
-        [LoggedTestMethod, Timeout(ConnectionStateChangeTestTimeoutMilliseconds)]
-        public async Task IotHubDeviceClient_DeviceDeleted_Gives_ConnectionStatus_DeviceDisabled_AMQP_WS()
+        [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
+        public async Task IotHubDeviceClient_DeviceDeleted_Gives_ConnectionStatus_Disconnected_ChangeReason_DeviceDisabled_AmqpWs()
         {
-            await IotHubDeviceClient_Gives_ConnectionStatus_DeviceDisabled_Base(
+            await IotHubDeviceClient_Gives_ConnectionStatus_Disconnected_ChangeReason_DeviceDisabled_Base(
                 new IotHubClientAmqpSettings(IotHubClientTransportProtocol.WebSocket),
                 async (r, d) => await r.Devices.DeleteAsync(d).ConfigureAwait(false)).ConfigureAwait(false);
         }
 
         [LoggedTestMethod, Timeout(ConnectionStateChangeTestTimeoutMilliseconds)] // This test always takes more than 5 minutes for service to return. Needs investigation.
         [TestCategory("LongRunning")]
-        public async Task IotHubDeviceClient_DeviceDisabled_Gives_ConnectionStatus_DeviceDisabled_AmqpTcp()
+        public async Task IotHubDeviceClient_DeviceDisabled_Gives_ConnectionStatus_Disconnected_ChangeReason_DeviceDisabled_AmqpTcp()
```

Comment:
It looks like both deleting and disabling devices on IoT hub now gives the same response -> Disconnected with DeviceDisabled. This seems a bit off considering now we don't have any way to help the user understand if their device was indeed disabled or deleted.
I'm not sure what the previous behavior was, but I think this is something we can improve. What's the error code from service on both of these operations?

Reply:
I will look into this and try to bring improvements in another PR